### DISCUSSION TITLE
How do I Fix the Following Warnings

### DISCUSSION THREAD

**Discussion by Ermiasmarino** [Homebrew/discussions#811] [Everyday usage]
I had tried to install Homebrew on HIgh Sierra and had error as a result. Then I upgraded to Mojave and Homebrew was installed without a problem. Now, however, I can't install applications using Homebrew because of directory ownership and writing errors. I also can't uninstall brew because brew gets uninstalled only partially in addition to getting "Operation not permitted" problems.

**Here is what brew doctor returned:** 

Last login: Sun Feb 14 16:04:26 on ttys000
Ermiass-MacBook-Pro:~ ermmy$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The following directories are not writable by your user:
/usr/local/lib/python3.7/site-packages

You should change the ownership of these directories to your user.
  sudo chown -R $(whoami) /usr/local/lib/python3.7/site-packages

And make sure that your user has write permission.
  chmod u+w /usr/local/lib/python3.7/site-packages

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libavcodec.55.52.102.dylib
  /usr/local/lib/libavformat.55.33.100.dylib
  /usr/local/lib/libavutil.52.66.100.dylib
  /usr/local/lib/libcgunwind.1.0.dylib
  /usr/local/lib/liblua.5.1.5.dylib
  /usr/local/lib/libswscale.2.5.102.dylib
  /usr/local/lib/libx264.129.dylib

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python
  vde
  qemu
  libssh
  p11-kit

Warning: Your Xcode (10.1) is too outdated.
Please update to Xcode 11.3.1 (or delete it).
Xcode can be updated from the App Store.

Warning: Your Xcode (10.1) is outdated.
Please update to Xcode 11.3.1 (or delete it).
Xcode can be updated from the App Store.

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install gmp jpeg libffi libunistring

Run `brew missing` for more details.

**Here is what I have done and what I got:** 

I have entered "sudo chown -R $(whoami) /usr/local/lib/python3.7/site-packages" and I got "Operation not permitted" errors. 

I have entered "chmod u+w /usr/local/lib/python3.7/site-packages" and nothing happens probably because I haven't taken ownership first. 

I haven't deleted the unbrewd dylibs because they are used by other applications. 

I have tried to "brew link" paython, vde, qemu, libssh, and p11-kit, but I got "Operation not permitted" errors such as "Linking /usr/local/Cellar/python/3.7.5... Error: Operation not permitted @ dir_s_rmdir - /usr/local/Frameworks/Python.framework/Versions"

I have tried "brew install gmp jpeg libffi libunistring", but I got the following error: 
Error: The following directories are not writable by your user:
/usr/local/lib/python3.7/site-packages

You should change the ownership of these directories to your user.
  sudo chown -R $(whoami) /usr/local/lib/python3.7/site-packages

And make sure that your user has write permission.
  chmod u+w /usr/local/lib/python3.7/site-packages

Please help me fix all these problems. If not, please help me uninstall and remove every trace of Homebrew from my computer(not just partially as mentioned above). Thanks.

**Comment by gromgit** [comment]
You've already asked for help on the same problem, have already received suggestions, and not responded to the last suggestion I gave: https://github.com/Homebrew/discussions/discussions/802

If what @carlocab and I wrote in that other thread is confusing, ask for more details in that thread instead of starting a new one.
